src/phase0/state_transition/epoch_processing.rs

```diff
@@ -745,9 +749,11 @@ pub fn get_source_deltas<
 ) -> Result<(Vec<Gwei>, Vec<Gwei>), Error> {
     // Return attester micro-rewards/penalties for source-vote for each validator.
     let previous_epoch = get_previous_epoch(state, context);
-    let matching_source_attestations =
-        get_matching_source_attestations(state, previous_epoch, context)?;
-    get_attestation_component_deltas(state, matching_source_attestations, context)
+    let mut matching_source_attestations = Vec::new();
+    for a in get_matching_source_attestations(state, previous_epoch, context)?.iter() {
+        matching_source_attestations.push(a);
+    }
```

Comment:
a bit easier to see what is going on w/ `collect` imo

`get_matching_source_attestations(...)?.iter().collect::<Vec<_>>()`